app/assets/javascripts/controllers/CommonChatController.ts

```diff
@@ -491,19 +491,20 @@ export default class CommonChatController {
             this.state.escalated = this.state.isEscalated();
         }
 
```

Comment:
Moved this._moveToClosingState() to the conditionals as it was changing the ChatStates so state.escalated would not be truthy. This was causing it to run _sendPostChatSurveyDigitalAssistant() even for webchats rather than _sendPostChatSurveyWebchat().